index.d.ts

```diff
@@ -19,6 +29,117 @@ export interface MonacoTailwindcssOptions {
    * worker.
    */
   tailwindConfig?: TailwindConfig | string
+  /**
+   * Extend Intellisense's settings.
+   * 
+   * Default values are based on the VS Code extension.
+   * 
+   * ``` typescript
+   * const defaultSettings = {
+      editor: {tabSize: 2},
+      tailwindCSS: {
+        emmetCompletions: false,
+        classAttributes: ["class", "className", "ngClass"],
+        codeActions: true,
+        hovers: true,
+        suggestions: true,
+        validate: true,
+        colorDecorators: true,
+        rootFontSize: 16,
+        showPixelEquivalents: true,
+        includeLanguages: {},
+        files: {exclude: []},
+        experimental: {
+          classRegex: [],
+          configFile: {}
+        },
+        lint: {
+          cssConflict: "warning",
+          invalidApply: "error",
+          invalidScreen: "error",
+          invalidVariant: "error",
+          invalidConfigPath: "error",
+          invalidTailwindDirective: "error",
+          recommendedVariantOrder: "warning"
+        },
+      }
+    }
+   */
+  intellisense?: Partial<{
+    editor: {
+      tabSize: number
+    }
+    tailwindCSS: Partial<{
+      emmetCompletions: boolean
+      includeLanguages: Record<string, string>
+      classAttributes: string[]
+      suggestions: boolean
+      hovers: boolean
+      codeActions: boolean
+      validate: boolean
+      showPixelEquivalents: boolean
+      rootFontSize: number
+      colorDecorators: boolean
+      files: { exclude: string[] }
+      experimental: {
+        classRegex: string[]
+        configFile: string | Record<string, string | string[]>
+      }
+      lint: {
+        cssConflict: DiagnosticSeveritySetting
+        invalidApply: DiagnosticSeveritySetting
+        invalidScreen: DiagnosticSeveritySetting
+        invalidVariant: DiagnosticSeveritySetting
+        invalidConfigPath: DiagnosticSeveritySetting
+        invalidTailwindDirective: DiagnosticSeveritySetting
+        recommendedVariantOrder: DiagnosticSeveritySetting
+      }
+    }>
+  }>
+  /**
+ * A way to call Tailwind's built-in Plugin API within the worker.
+ * ``` typescript
+ * 
+   // Instead of calling the functions inside your config like this...
+ * const tailwindConfig = {
+   // ...config
+ *  plugins: [
+      ({addUtilities}) => {
+        addUtilities({
+          '.custom-class': {
+            color: '#000000',
+            fontSize: '1rem',
+            fontWeight: 900
+          }
+        })
+      }
+    ]
+ * }
+
+// ...you provide an array containing the arguments
+// for each function you wish to call.
```

Comment:
This is your interpretation of how the configuration could work. This might cater to your specific situation, but it might not to others’. The `prepareTailwindConfig` option from `initialize` allows you to fully customize how the Tailwind config provided could be interpreted.

Reply:
Alright, this whole thing was a big overlook, might call it a skill issue, on my side... Remember in the issue that led to this PR when I said I tried `initialize` but didn't make it work and couldn't really understand what is its purpose?

I was calling that thing just above `window.MonacoEnvironment` where I create the different workers... This is also my first time using workers, I didn't know better. Now, I've created a file for the worker, use `initialize` in it and point the `tailwindcss` worker to that file. And it just works...